windows/RNSVG/SvgView.cpp

```diff
@@ -189,16 +221,54 @@ void SvgView::Unload() {
   if (m_group) {
     m_group.Unload();
   }
-
-  if (m_canvas) {
-    m_canvas.RemoveFromVisualTree();
-    m_canvas = nullptr;
-  }
 }
 
-void SvgView::InvalidateCanvas() {
-  if (m_hasRendered) {
-    m_canvas.Invalidate();
+void SvgView::Invalidate() {
+  if (!m_loaded) {
+    return;
   }
+
+  m_brushes.Clear();
+  m_templates.Clear();
+
+  com_ptr<ID2D1DeviceContext> deviceContext{get_self<D2DDeviceContext>(DeviceContext())->Get()};
+
+  Size size{static_cast<float>(ActualWidth()), static_cast<float>(ActualHeight())};
+
+  xaml::Media::Imaging::SurfaceImageSource surfaceImageSource(
+      static_cast<int32_t>(size.Width), static_cast<int32_t>(size.Height));
+  com_ptr<ISurfaceImageSourceNativeWithD2D> sisNativeWithD2D{surfaceImageSource.as<ISurfaceImageSourceNativeWithD2D>()};
+
+  // Associate the Direct2D device with the SurfaceImageSource.
+  com_ptr<ID2D1Device> device{get_self<D2DDevice>(Device())->Get()};
+  sisNativeWithD2D->SetDevice(device.get());
+
+  com_ptr<IDXGISurface> dxgiSurface;
+
+  RECT updateRect{0, 0, static_cast<long>(size.Width), static_cast<long>(size.Height)};
+  POINT offset{0, 0};
+  check_hresult(sisNativeWithD2D->BeginDraw(updateRect, __uuidof(IDXGISurface), dxgiSurface.put_void(), &offset));
+
+  // Create render target.
+  com_ptr<ID2D1Bitmap1> bitmap;
+  check_hresult(deviceContext->CreateBitmapFromDxgiSurface(dxgiSurface.get(), nullptr, bitmap.put()));
+
+  // Set context's render target.
+  deviceContext->SetTarget(bitmap.get());
+
+  // Draw using Direct2D context
+  deviceContext->BeginDraw();
+
+  auto transform = Numerics::make_float3x2_translation({static_cast<float>(offset.x), static_cast<float>(offset.y)});
+  deviceContext->SetTransform(D2DHelpers::AsD2DTransform(transform));
+
+  deviceContext->Clear(D2D1::ColorF(D2D1::ColorF::Orange, 0.0f));
```

Comment:
Clear color is for debug purposes, right? Should this be removed?

Reply:
Orange was for debug purposes. I'll change it to the more usual Black (though it's transparent), but it does need to clear the devicecontext before drawing.